src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorCodeDocumentExtensions.cs

```diff
@@ -555,22 +555,11 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
         {
             if (node != null && node.DirectiveDescriptor == NamespaceDirective.Directive)
             {
-                var directiveContent = node.Body?.GetContent();
-
-                // In practice, this should never be null and always start with 'namespace'. Just being defensive here.
-                if (directiveContent != null && directiveContent.StartsWith(NamespaceDirective.Directive.Directive, StringComparison.Ordinal))
+                var children = node.Body?.ChildNodes().ToImmutableArray();
+                if (children is [_, CSharpCodeBlockSyntax {  Children: [ _, CSharpSyntaxNode @namespace, ..] }])
```

Comment:
We know its a namespace because of the `DirectiveDescriptor` check above. I'm not sure it's worth asserting anything here; we're using the layout of what is parsed so it shouldn't be possible for it to be different. If we ever change the parsing then the namespace tests would start failing anyway, so I don't think it buys us anything.